core/src/consensus/tendermint/worker.rs

```diff
@@ -1456,7 +1456,7 @@ impl Worker {
                     self.save_last_confirmed_view(prev_block_view);
                 }
             }
-            if height_changed {
+            if height_changed && (self.height, Step::Propose) > (self.height, self.step.to_step()) {
```

Comment:
Why is the self.height needed?
It seems the two expressions are the same.
1. (self.height, Step::Propose) > (self.height, self.step.to_step())
2. Step::Propose > self.step.to_step()